### PR TITLE
8286560: Remove user parameter from jdk.internal.perf.Perf.attach()

### PR DESCRIPTION
**Pull request by iklam**
The API `jdk.internal.perf.Perf.::attach(String user, int lvmid)` is never used. It should be removed, and all the handling of a specified user name should be removed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286560](https://bugs.openjdk.java.net/browse/JDK-8286560): Remove user parameter from jdk.internal.perf.Perf.attach()

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [afa66232](https://git.openjdk.java.net/jdk/pull/8669/files/afa6623224f1afb54b948326834bb6c16921784a)
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to [afa66232](https://git.openjdk.java.net/jdk/pull/8669/files/afa6623224f1afb54b948326834bb6c16921784a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8669/head:pull/8669` \
`$ git checkout pull/8669`

Update a local copy of the PR: \
`$ git checkout pull/8669` \
`$ git pull https://git.openjdk.java.net/jdk pull/8669/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8669`

View PR using the GUI difftool: \
`$ git pr show -t 8669`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8669.diff">https://git.openjdk.java.net/jdk/pull/8669.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Nice cleanup! I checked back in JDK 7 and couldn't find any use of this particular API.

Thanks.

File: src/java.base/share/classes/jdk/internal/perf/Perf.java [hunk 1]
```diff
@@ -298,7 +242,7 @@ public void run() {
      * @throws  OutOfMemoryError The instrumentation buffer could not be mapped
      *                           into the virtual machine's address space.
      */
-    private native ByteBuffer attach0(String user, int lvmid)
+    private native ByteBuffer attach0(int lvmid)
                    throws IllegalArgumentException, IOException;
```
Comment (AlanBateman) [inline src/java.base/share/classes/jdk/internal/perf/Perf.java:246]: You can drop the "throws IllegalArgumentException" here if you want, it's not needed as it's a runtime exception.

**Review by AlanBateman — APPROVED** [review]

File: src/java.base/share/classes/jdk/internal/perf/Perf.java (hunk 1 above)
Comment (iklam) [reply src/java.base/share/classes/jdk/internal/perf/Perf.java]: Fixed.

**Comment by iklam** [comment]
Thanks to @dholmes-ora and @AlanBateman for the review.
/integrate

**Pull request closed without merging**